qiita_pet/support_files/doc/source/dev/plugins.rst

```diff
@@ -0,0 +1,81 @@
+.. _plugins:
+
+.. index :: plugins
+
+Plugins
+=======
+
+Qiita plugins allow to extend the functionality of Qiita by adding new data
+formats and/or new processing pipelines. The list of officially supported
+plugins can be found in the `Qiita Spots <https://github.com/qiita-spots>`__
+GitHub organization.
+
+Overview of Qiita's plugin system
+---------------------------------
+
+Qiita's plugin system has been designed to increase Qiita's flexibility to
+process -omics datasets, while allowing plugin developers to create plugins
+using the programming language that they are more comfortable with.
+
+In order to achieve this flexibility, Qiita represents the different -omics
+datasets using ``Artifacts``. An ``Artifact`` is a file or set of files that
+conceptually represents a dataset. However, Qiita does not understand the
+contents of the artifact, so it doesn't know to show information about the
+artifact or which pipelines/methods can be applied to it. It is up to the Qiita
+plugins to define what an artifact is and which methods can be applied to it.
+With the goal of reducing community development effort, there are two different
+types of plugins: ``Qiita Type Plugins`` and ``Qiita Plugins``.
+
+The ``Qiita Type Plugins`` define new artifact types. These plugins must define
+two operations: ``Validate`` and ``Generate HTML summary``.
```

Comment:
Perhaps worth adding the name of the file where they are defined and even the line where the TODO is.